File: src/Educai.Domain/Entities/UserPreference.cs
namespace Educai.Domain.Entities;

public class UserPreference
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    
    //hedefler
    public string GoalSubject { get; set; } = "Genel";
    public int DailyTarget { get; set; } = 20;
    public int WeeklyQuestionTarget { get; set; } = 100;
    public List<string> FocusAreas { get; set; } = new();
    
    //navigation property
    public User User { get; set; } = null!;
}

File: src/Educai.Web/src/types/UserPreference.ts
export interface UserPreference {
  id: string
  userId: string
  goalSubject: string
  dailyTarget: number
  weeklyQuestionTarget: number
  focusAreas: string[]
}

export interface CreateUserPreferencePayload {
  userId: string
  goalSubject: string
  dailyTarget: number
  weeklyQuestionTarget: number
  focusAreas: string[]
}

export interface UpdateUserPreferencePayload {
  goalSubject: string
  dailyTarget: number
  weeklyQuestionTarget: number
  focusAreas: string[]
}
